### PR TITLE
Move to Sphinx documentation

### PR DESCRIPTION
**Pull request by chsasank**
As discussed in #640 

Markdown files are mostly left the same. Only difference is there's index.rst in place of index.md. 

How to build

```
cd docs/
make html
# open _build/html/index.html
```

How it looks:

image

### CODE REVIEW

**Comment by chsasank** [comment]
I think the PR is complete and ready to merge :)

Preview the changes by extracting this zip and opening index.html

[html.zip](https://github.com/lark-parser/lark/files/5080556/html.zip)

File: lark/lark.py [hunk 1]
```diff
@@ -295,6 +316,7 @@ def __init__(self, grammar, **options):
             with FS.open(cache_fn, 'wb') as f:
                 self.save(f)
 
+    # TODO: merge with above
```
Comment (chsasank) [inline lark/lark.py:319]: @erezsh need guidance here. Are you use you want to keep this docstring? If so, it'll show up twice in the docs, which is not optimal

Comment (chsasank) [reply lark/lark.py:319]: You can instead refer to help(lark.lark.LarkOptions) in the docstring of options.

**Comment by erezsh** [comment]
`make.bat` doesn't seem to work:

```bash
PS C:\code\lark\docs> .\make.bat
Sphinx v2.3.1
Please use `make target' where target is one of
  html        to make standalone HTML files
  dirhtml     to make HTML files named index.html in directories
  singlehtml  to make a single large HTML file
  pickle      to make pickle files
  json        to make JSON files
  htmlhelp    to make HTML files and an HTML help project
  qthelp      to make HTML files and a qthelp project
  devhelp     to make HTML files and a Devhelp project
  epub        to make an epub
  latex       to make LaTeX files, you can set PAPER=a4 or PAPER=letter
  text        to make text files
  man         to make manual pages
  texinfo     to make Texinfo files
  gettext     to make PO message catalogs
  changes     to make an overview of all changed/added/deprecated items
  xml         to make Docutils-native XML files
  pseudoxml   to make pseudoxml-XML files for display purposes
  linkcheck   to check all external links for integrity
  doctest     to run all doctests embedded in the documentation (if enabled)
  coverage    to run coverage check of the documentation (if enabled)
```

File: lark/lark.py (hunk 1 above)
Comment (erezsh) [reply lark/lark.py:319]: LarkOptions is not the interface, it's a helper class. It doesn't really belong in the documentation, except for maybe a short comment.

**Comment by erezsh** [comment]
It asked me pip install `recommonmark` and `sphinx_rtd_theme`.

Won't that be a problem on readthedocs?

**Comment by chsasank** [comment]
I'm pretty sure they're installed by default on read the docs. Both these packages are from read the docs themselves. If the build failed, we have to install them using the following guide: https://docs.readthedocs.io/en/stable/guides/specifying-dependencies.html

File: lark/lark.py (hunk 1 above)
Comment (chsasank) [reply lark/lark.py:319]: Makes sense. Will change the docs based on this.

**Comment by erezsh** [comment]
You deleted this part from the documentation: https://lark-parser.readthedocs.io/en/latest/classes/#using-unicode-character-classes-with-regex

**Comment by chsasank** [comment]
> You deleted this part from the documentation: https://lark-parser.readthedocs.io/en/latest/classes/#using-unicode-character-classes-with-regex

Thanks added. Ready to merge. Will be on standby for next 30 mins so that merge and deployment of docs is smooth.

**Pull request merged**